### PR TITLE
fix(docs): replace twitter logo/links with x logo/links

### PR DESCRIPTION
**Pull request by dgz9**
Closes # N/A

## 📝 Description

> This PR updates all Twitter references to X (formerly Twitter) in the NextUI documentation. This includes changing the Twitter logo to the X logo and updating all twitter.com links to x.com.

## ⛳️ Current behavior (updates)

> Currently, the NextUI documentation uses the old Twitter bird logo and links to twitter.com for social media references.

## 🚀 New behavior

> - The Twitter bird logo has been replaced with the new X logo.
> - All twitter.com links have been updated to x.com.
> - The TwitterIcon component has been renamed to XIcon and its SVG path has been updated to represent the X logo.

## 💣 Is this a breaking change (Yes/No):

> No

## 📝 Additional Information

> N/A

## Summary by CodeRabbit

- **New Features**
	- Introduced a new `XIcon` component for the updated branding.
  
- **Bug Fixes**
	- Updated various URLs from Twitter to X to reflect the platform rebranding.

- **Documentation**
	- Revised links in README and other documentation to point to the new X domain. 

- **Style**
	- Adjusted the visual presentation of the `LinkIcon` in the `FeaturesGrid` component by removing the specific class name.

### CODE REVIEW

**Review by wingkwong — CHANGES_REQUESTED** [review]
`TwitterIcon` can be removed since it's not in use.

**Comment by dgz9** [comment]
> `TwitterIcon` can be removed since it's not in use.

Will do tonight.

Do you want me to remove all references of Twitter and replace with X or just keep it at the logo and links?

**Comment by wingkwong** [comment]
I mean just `TwitterIcon` in  `apps/docs/components/icons/social.tsx` since you replaced it with `XIcon`. `TwitterIcon` is not in use anymore.

**Comment by dgz9** [comment]
I was wondering about all the references to twitter, but I left it for now.

I removed the `TwitterIcon` and changed a few more name references (even though it looks weird as just X).

**Comment by wingkwong** [comment]
Is it ready for review or you still have further changes?

**Comment by dgz9** [comment]
Ready for review. Let me know if you want any other changes.

File: apps/docs/components/navbar.tsx [hunk 1]
```diff
@@ -325,12 +325,12 @@ export const Navbar: FC<NavbarProps> = ({children, routes, mobileRoutes = [], sl
         <NavbarItem className="hidden sm:flex">
           <Link
             isExternal
-            aria-label="Twitter"
+            aria-label="X"
             className="p-1"
             href={siteConfig.links.twitter}
-            onPress={() => handlePressNavbarItem("Twitter", siteConfig.links.twitter)}
+            onPress={() => handlePressNavbarItem("X", siteConfig.links.twitter)}
```
Comment (wingkwong) [inline apps/docs/components/navbar.tsx:331]: This change would change the trace event title in analytics. please rollback.

File: .changeset/slimy-cats-brush.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+"@nextui-org/user": patch
+"@nextui-org/react": patch
```
Comment (wingkwong) [inline .changeset/slimy-cats-brush.md:3]: since the changes don't involve version bump, you may remove these two lines.

**Review by wingkwong — CHANGES_REQUESTED** [review]
final round of changes

**Review by wingkwong — CHANGES_REQUESTED** [review]
X icon in light theme is not visible.

image
image

**Comment by dgz9** [comment]
Fixed light theme icon. Also noticed the feature grid icon was not working for light more, changed that too.

**Review by wingkwong — APPROVED** [review]

**Pull request merged**